compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -1765,82 +1768,97 @@ private Result seekInAccessExpression(ParserRuleContext currentCtx, int lookahea
      */
     private Result seekMatchInExpressionRhs(int lookahead, int currentDepth, int currentMatches, boolean isEntryPoint) {
         ParserRuleContext parentCtx = getParentContext();
-        ParserRuleContext[] next;
+        if (parentCtx == ParserRuleContext.FUNCTION_CALL_START) {
+            endContext();
+            return seekMatchInExpressionRhs(lookahead, currentDepth, currentMatches, isEntryPoint, true);
+        } else {
+            return seekMatchInExpressionRhs(lookahead, currentDepth, currentMatches, isEntryPoint, false);
+        }
+    }
+
+    private Result seekMatchInExpressionRhs(int lookahead, int currentDepth, int currentMatches, boolean isEntryPoint,
+                                            boolean allowFuncCall) {
+        ParserRuleContext parentCtx = getParentContext();
+        ParserRuleContext[] alternatives = null;
         switch (parentCtx) {
             case ARG_LIST:
-                next = new ParserRuleContext[] { ParserRuleContext.BINARY_OPERATOR, ParserRuleContext.DOT,
+                alternatives = new ParserRuleContext[] { ParserRuleContext.BINARY_OPERATOR, ParserRuleContext.DOT,
                         ParserRuleContext.ANNOT_CHAINING_TOKEN, ParserRuleContext.OPTIONAL_CHAINING_TOKEN,
                         ParserRuleContext.CONDITIONAL_EXPRESSION, ParserRuleContext.XML_NAVIGATE_EXPR,
                         ParserRuleContext.MEMBER_ACCESS_KEY_EXPR, ParserRuleContext.COMMA,
                         ParserRuleContext.ARG_LIST_START, ParserRuleContext.ARG_LIST_END };
-                return seekInAlternativesPaths(lookahead, currentDepth, currentMatches, next, isEntryPoint);
+                return seekInAlternativesPaths(lookahead, currentDepth, currentMatches, alternatives, isEntryPoint);
```

Comment:
Under `ARG_LIST` context, `ARG_LIST_START` is always there as an alternative.
In other contexts, `ARG_LIST_START` is added as an alternative only if `allowFuncCall` is `true`.

Reply:
Why is that needed?
What if is write `expr()` inside func-args?